java-spiffe-core/src/main/java/spiffe/bundle/jwtbundle/JwtBundleSet.java

```diff
@@ -0,0 +1,73 @@
+package spiffe.bundle.jwtbundle;
+
+import lombok.NonNull;
+import lombok.Value;
+import lombok.val;
+import spiffe.bundle.BundleSource;
+import spiffe.exception.BundleNotFoundException;
+import spiffe.spiffeid.TrustDomain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A <code>JwtBundleSet</code> represents a set of JWT bundles keyed by trust domain.
+ */
+@Value
+public class JwtBundleSet implements BundleSource<JwtBundle> {
+
+    ConcurrentHashMap<TrustDomain, JwtBundle> bundles;
+
+    private JwtBundleSet(Map<TrustDomain, JwtBundle> bundles) {
+        this.bundles = new ConcurrentHashMap<>(bundles);
+    }
+
+    /**
+     * Creates a JWT bundle set from the list of JWT bundles.
+     *
+     * @param bundles List of {@link JwtBundle}
+     * @return a {@link JwtBundleSet}
+     */
+    public static JwtBundleSet of(@NonNull final List<JwtBundle> bundles) {
+        Map<TrustDomain, JwtBundle> bundleMap = new HashMap<>();
```

Comment:
Can also be done in one statement with the Stream API:

```java
Map<TrustDomain, JwtBundle> bundleMap = bundles.stream()
    .collect(Collectors.toMap(
        b -> b.getTrustDomain(),
        Function.identity(),
       (existing, replacement) -> replacement));
```

Reply:
IMO in this case the functional style feels obscure, I prefer to do it in 2 lines using the imperative style, I think it shows clearly the intention of the code.